archive-node: configuration for old networks (#836)
oasisprotocol/docs: docs/node/run-your-node/archive-node.md

## Prerequisites

> Running an archive node requires a pre-existing oasis-node state.

Reviewer: ```suggestion
Running an archive node requires a pre-existing `oasis-node` state.
```

> ## Configuration (v23 onward)

Reviewer: Not sure, if we should either stick to the upgrade names or the Oasis Core versions. In any case, the titles should be consistent IMO.

```suggestion
## Configuration (Oasis Core 23 and later)
```

> Starting from version 23, the configuration for enabling archive mode has changed. Use the mode

Reviewer: ```suggestion
Starting from the Oasis Core version 23, the configuration for enabling archive mode has changed. Use the `mode`
```

## Prerequisites

> Running an archive node requires a pre-existing `oasis-node` state.

Reviewer: What does this mean, if you wanted to spin up an archive node from scratch? Will it sync automatically from other archive nodes or do you need to download the state snapshot from somewhere?

Author: No, you need to have state from somewhere.

Reviewer: Maybe extend with something like this:

```suggestion
Running an archive node requires a pre-existing `oasis-node` state. If you don't have one,
you can download a snapshot of a specific network state [here][snapshots].

[snapshots]: https://snapshots.oasis.io
```

Author: thanks, added

:::info

> Keep all other settings the same as those for a full node. For example, to serve archived runtime

Reviewer: I would rather have a complete Client config pasted above with the archive mode enabled, so one can simply set up their own archive node from scratch when reading the chapter. You don't need SGX, a configured entity or being elected in the validator for running the archive node right?

Author: Ok, i updated the configs. Need to double check if the configs are correct, but I believe these should work, since i picked them from docs git history.

state, the node needs to have the runtime configured and the state present.

:::

> ## Configuration (pre-Eden)

Reviewer: ```suggestion
## Configuration (Oasis Core 22 and earlier)
```

For all pre-Eden networks, such as Damask, the configuration remains the same but requires the

> appropriate version of `oasis-node` and node state.

Reviewer: ```suggestion
appropriate version of `oasis-node` and the node state.
```

#### Damask

> To run an archive node for Damask, use [Oasis Node v22.2.12] and the following

Reviewer: ```suggestion
To run an archive node for Damask, use [Oasis Core 22.2.12] and the following
```

configuration:

```yaml
# other settings omitted ...
consensus:
  tendermint:
    mode: archive
```

#### Cobalt

> To run an archive node for Cobalt, use [Oasis Node v21.3.14] and the following configuration:

Reviewer: ```suggestion
To run an archive node for Cobalt, use [Oasis Core 21.3.14] and the following configuration:
```

```yaml
# other settings omitted ...
consensus:
  tendermint:
    mode: archive
```

:::warn

> Ensure you are using the correct version of oasis-node and pre-existing state for your specific pre-Eden network.

Reviewer: Also trim to 80 characters.

```suggestion
Ensure you are using the correct version of oasis-node and the pre-existing state for your specific pre-Eden network.
```

## See also

[Archive Web3 Gateway](../web3.mdx#archive-web3-gateway)

> [Oasis Node v22.2.12]: https://github.com/oasisprotocol/oasis-core/releases/tag/v22.2.12
> [Oasis Node v21.3.14]: https://github.com/oasisprotocol/oasis-core/releases/tag/v21.3.14

Reviewer: ```suggestion
[Oasis Core 22.2.12]: https://github.com/oasisprotocol/oasis-core/releases/tag/v22.2.12
[Oasis Core 21.3.14]: https://github.com/oasisprotocol/oasis-core/releases/tag/v21.3.14
```
